aws_lambda_powertools/utilities/parser/parser.py

```diff
@@ -76,11 +77,20 @@ def handler(event: Order, context: LambdaContext):
     ValidationError
         When input event does not conform with model provided
     InvalidModelTypeError
-        When model given does not implement BaseModel
+        When model given does not implement BaseModel or is not provided
     InvalidEnvelopeError
         When envelope given does not implement BaseEnvelope
     """
-    parsed_event = parse(event=event, model=model, envelope=envelope) if envelope else parse(event=event, model=model)
+    if model is None:
+        type_hints = typing.get_type_hints(handler)
+        model = type_hints.get("event")
+        if model is None:
+            raise InvalidModelTypeError(
+                "The model must be provided either as the `model` argument to `event_parser`"
+                "or as the type hint of `event` in the handler that it wraps",
+            )
```

Comment:
I understand why you're failing here when neither the model nor type hint is defined, but we've already tested it here: https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/utilities/parser/parser.py#L167. I'm not sure if we need to add this here, I think it's better change our documentation to inform this. What do you thing?

I have a suggestion to make this code even simpler:
```suggestion
    model = model or typing.get_type_hints(handler).get("event")
```

Reply:
I was wrong about that experience. It actually makes sense to fail early and show a specific message when the customer doesn't provide the model or the event.